### ISSUE TITLE
New BSIP: Design a Primary market which can map the orders of the Gateway market

### ISSUE THREAD

**Issue by shulthz** [bitshares/bsips#245]
**Cause**
Inter-Blockchain technology still need much time to research and develop, but we still need to design a market first for these functions or other.
BTS has so many gateway markets, but can't make a well depth and liquidity for each other and users.
The target audience was limited for the gateway market.

**GOAL**
Design a market first for these functions or other.
Solve the decentralized problem of gateway market, give the user more better trading experience, they don't need to face the poor depth and liquidity of the gateway market which they only know.
Give the gateway a more wider audience, increase the depth and liquidity.

**Scheme**
Design a Primary market: this market will map the orders of the Gateway market.
In the Primary market:
Taker can fill the orders which were mapped to the Primary market; the real trade happend in the Gateway market.
Taker can fill the pending orders which were directly placed in the the Primary market; the real trade happend in the Primary market.
The orders which happend in the Gateway market directly will can't influence the maker in the Primary market;

**Problem**
Trust problem of Gateway;
Market fee problem;
Withdraw problem;
Market access conditions;
...

**Example**
![图片](https://user-images.githubusercontent.com/34892308/67171332-31350a80-f3e9-11e9-8c0a-e2ecf51449dc.png)

**Comment by pmconrad** [comment]
Hm, interesting. This could even work if there was a mechanism for exchanging the "Primary BTC" tokens for any of the gateway tokens. Something like Bancor would be useful...

**Comment by shulthz** [comment]
> 
> 
> Hm, interesting. This could even work if there was a mechanism for exchanging the "Primary BTC" tokens for any of the gateway tokens. Something like Bancor would be useful...

The Fiat price of BTC have a little difference in some areas.
But price of the bts/btc didn't have this difference.

**Comment by froooze** [comment]
@shulthz 
The idea is good, but the problems are bigger. I see three options for this:
- a smart-multi-UIA-coin based on existing UIAs, circulating supply of different UIAs defines proportion 
- increase liquidity for OPEN.BTC/GDEX.BTC and other arbitrage markets + include order matching on the first level (bots will be faster anyways)
- integrate  trustless  BTC gateways

>  Something like Bancor would be useful...

Yes, something Bancor Protocol -> Uniswap -> [MLP](#213 )

**Comment by pmconrad** [comment]
> The Fiat price of BTC have a little difference in some areas.

Yes, but the gateway UIAs are not the same as actual BTC. E. g. one gateway may have KYC requirements that make it unusable for some users, while others don't. Or a gateway may have technical problems and disable withdrawals. If a gateway goes broke, its UIA will probably be worth much less.

**Comment by ioBanker** [comment]
Thanks for taking time in drafting your idea.

I don’t see this working for UIAs because the concept behind UIAs is to let users create their own means of these UIA, core code should never involve itself to facilitate a proxy for gateways nor particular UIA, core should stay as enabler for gateways and should not be a part of any legal responsibilities related to gateways operations, this solution would enforce users to buy from a gateway that they might not trust when the markets of assets are merged that way.

**Comment by shulthz** [comment]
> 
> 
> Thanks for taking time in drafting your idea.
> 
> I don’t see this working for UIAs because the concept behind UIAs is to let users create their own means of these UIA, core code should never involve itself to facilitate a proxy for gateways nor particular UIA, core should stay as enabler for gateways and should not be a part of any legal responsibilities related to gateways operations, this solution would enforce users to buy from a gateway that they might not trust when the markets of assets are merged that way.

Yes, it's the trust problem of Gateway, but this didn't mean to force the users to trade in the Primary market, it was designed for the people who like trade in the Primary market, the user who like trade in the Gateway market still can trade in there.

The initially thought: The BTC trade in the Primary market still is the gateway.BTC, just show the name is BTC in this trading pair, these BTC will still show the name of gateway.BTC in the Portfolio.

This market wasn't designed for the gateway only, it will have many purposes.
E.G. 
1. Planning a market making contest;
2. Prepare for integrate trustless gateways or Inter-Blockchain technology;
...
The problem still is :
Trust problem of Gateway;
Market access conditions;
